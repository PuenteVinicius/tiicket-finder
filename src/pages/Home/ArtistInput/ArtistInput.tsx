import React, { useState } from "react";
import { TextInput, Box } from "grommet";

const ArtistInput = () => {
  const [artistNameState, setArtistNameState] = useState({ artistName: "" });

  const artistNameChangeHandler = (event: any) => {
    const changedArtistName: string = event.target.value;
    setArtistNameState({ artistName: changedArtistName });
  };

  return (
    <TextInput
      data-testid="artist-input"
      placeholder="FaÃ§a sua busca aqui..."
      onChange={artistNameChangeHandler}
    />
  );
};

export default ArtistInput;
