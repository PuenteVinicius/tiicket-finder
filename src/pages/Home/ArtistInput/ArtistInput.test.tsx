import { render, fireEvent, screen } from "@testing-library/react";
import "jest-styled-components";
import ArtistInput from "./ArtistInput";

describe("ArtistInput component", () => {
  let artistInput: any;
  beforeEach(() => {
    artistInput = render(<ArtistInput />);
  });
  it("Should render text correctly", () => {
    expect(
      artistInput.getByPlaceholderText("FaÃ§a sua busca aqui...")
    ).toBeInTheDocument();
  });

  it("renders home correctly", () => {
    expect(artistInput).toMatchSnapshot();
  });

  describe("Should be fired when an astist has changed", () => {
    it("When user type a band or an artist name", () => {
      let input = screen.getByTestId("artist-input");
      fireEvent.change(input, { target: { value: "Dua Lipa" } });
      expect(artistInput.getByDisplayValue("Dua Lipa")).toBeInTheDocument();
    });
  });
});
