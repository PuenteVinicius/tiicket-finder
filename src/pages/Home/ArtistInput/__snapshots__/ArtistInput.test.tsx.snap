// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistInput component renders home correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus > circle,
.c1:focus > ellipse,
.c1:focus > line,
.c1:focus > path,
.c1:focus > polygon,
.c1:focus > polyline,
.c1:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus::-moz-focus-inner {
  border: 0;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

.c0 {
  position: relative;
  width: 100%;
}

<body>
    <div>
      <div
        class="c0"
      >
        <input
          autocomplete="off"
          class="c1"
          data-testid="artist-input"
          placeholder="Faça sua busca aqui..."
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 cnyLED"
    >
      <input
        autocomplete="off"
        class="StyledTextInput-sc-1x30a0s-0 evYMFd"
        data-testid="artist-input"
        placeholder="Faça sua busca aqui..."
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
